lib/amqp/client/connection.rb

```diff
@@ -27,6 +27,7 @@ def close(reason = "", code = 200)
       write_bytes FrameBytes.connection_close(code, reason)
       expect(:close_ok)
       @closed = true
+      @socket.close
```

Comment:
Is this needed? Isn't the socket closed in the `read_loop` after a `close_ok` frame? Or maybe this is the better place to have it in?

Reply:
I think we added this together actually, not sure why, I'll remove it